Meziantou.FluentAssertionsAnalyzers/XUnitAssertAnalyzerCodeFixProvider.cs

```diff
@@ -152,6 +153,17 @@ private static async Task<Document> Rewrite(Document document, SyntaxNode nodeTo
                 MemberAccessExpression(InvokeShould(originalArguments[1]), newMethodName),
                 ArgumentList(originalMethod.ArgumentList.Arguments[0].Expression, originalMethod.ArgumentList.Arguments[2].Expression));
         }
+        else if (methodName is "Equal" or "NotEqual" && method.Parameters.Length == 3 && method.Parameters[2].Type.Name == nameof(IEqualityComparer))
```

Comment:
You must compare symbols correctly to avoid false-positive

```suggestion
        else if (methodName is "Equal" or "NotEqual" && method.Parameters.Length == 3 && method.Parameters[2].Type.Equals(compilation, "System.Collections.Generic.IEqualityComparer`1"))
```

Reply:
hm, this suggestion does not work. I thinking about it

Reply:
There is a similar check above, maybe it can give you an example. If it doesn't work, I can have a look at it later

https://github.com/meziantou/Meziantou.FluentAssertionsAnalyzers/pull/10/files/99e88cb5a7fba0f94013f8b96cbbf8f30ae75171#diff-82ee5eb7e03488eb286be3ef4e07f18a6728aab925f72ca5c64a886702bf4cffR101

Reply:
This worked with code:
```csharp
 else if (methodName is "Equal" or "NotEqual" && method.Parameters.Length == 3 && method.Parameters[2].Type.OriginalDefinition.IsOrImplements(compilation, "System.Collections.Generic.IEqualityComparer`1"))
```

Reply:
`Assert.Equal` is a generic method, so you need to get the original definition of the symbol to convert `IEqualityComparer<SpecificType>` to `IEqualityComparer<T>`. Then the comparison should work.

````
method.Parameters[2].Type.OriginalDefinition.Equals(...);
````